Make Docker build pass (problem with gcc failure, but could be more)

So that it doesn't crash and installs the requirements as expected.

Reproduced on Mac+Docker:
```
$ cd test-server && docker build -t test-server .
...
    src/blob.h:33:18: fatal error: git2.h: No such file or directory
    compilation terminated.
    error: command 'gcc' failed with exit status 1

    ----------------------------------------
Command "/usr/bin/python -u -c "import setuptools, tokenize;__file__='/tmp/pip-build-0BuNob/pygit2/setup.py';f=getattr(tokenize, 'open', open)(__file__);code=f.read().replace('\r\n', '\n');f.close();exec(compile(code, __file__, 'exec'))" install --record /tmp/pip-ZTC5e7-record/install-record.txt --single-version-externally-managed --compile" failed with error code 1 in /tmp/pip-build-0BuNob/pygit2/
The command '/bin/sh -c pip install -r /requirements.txt' returned a non-zero code: 1
```

I'll pay $15 to fix this under [ScalaWilliam Work rules](http://work.scalawilliam.com/rules/). Please say "**I'm taking on this**" to reserve this task to yourself. Posted also on: http://work.scalawilliam.com/dockerfile-requirements-build/

--- a/test-server/Dockerfile
+++ b/test-server/Dockerfile
@@ -2,6 +2,9 @@ FROM jkarlos/git-server-docker
 
 # https://github.com/gliderlabs/docker-alpine/blob/master/docs/usage.md
 RUN apk add --no-cache \
+  libxslt-dev \
+  libxml2-dev \
+  libgit2-dev \
   curl \
   python \
   python-dev \

--- a/test-server/requirements.txt
+++ b/test-server/requirements.txt
@@ -1,3 +1,3 @@
-pygit2
+pygit2==0.24.2
 lxml
 requests

